File: nomad-samples/js/asset-groups/asset-groups.js
import addAssetToAssetGroup from "./asset-groups/add-asset-to-asset-group.js";
import createAssetGroup from "./asset-groups/create-asset-group.js";
import deleteAssetGroup from "./asset-groups/delete-asset-group.js";
import getAssetGroups from "./asset-groups/get-asset-groups.js";
import removeAssetFromAssetGroup from "./asset-groups/remove-assets-from-asset-group.js";
import renameAssetGroup from "./asset-groups/rename-asset-group.js";

const AUTH_FORM = document.getElementById("authForm");
const GET_FORM = document.getElementById("getForm");
const CREATE_FORM = document.getElementById("createForm");
const ADD_FORM = document.getElementById("addForm");
const REMOVE_FORM = document.getElementById("removeForm");
const RENAME_FORM = document.getElementById("renameForm");
const DELETE_FORM = document.getElementById("deleteForm");

const TOKEN_INPUT = document.getElementById("authInput");
const ADD_INPUT = document.getElementById("addInput");
const ADD_ASSET_INPUT = document.getElementById("addAssetInput");
const REMOVE_INPUT = document.getElementById("removeInput");
const REMOVE_ASSET_INPUT = document.getElementById("removeAssetInput");
const RENAME_INPUT = document.getElementById("renameInput");
const DELETE_INPUT = document.getElementById("deleteInput");

sessionStorage.clear();

AUTH_FORM.addEventListener("submit", function (event)
{
    event.preventDefault();
    sessionStorage.setItem("token", TOKEN_INPUT.value);
    console.log("Successfuly updated token");
});

GET_FORM.addEventListener("submit", function (event)
{
    event.preventDefault();
    getForm();
});

CREATE_FORM.addEventListener("submit", function (event)
{
    event.preventDefault();
    createForm();
});

ADD_FORM.addEventListener("submit", function (event)
{
    event.preventDefault();

    let assetGroup = ADD_INPUT.value;
    let assets = ADD_ASSET_INPUT.value;

    addForm(assetGroup, assets);
});

REMOVE_FORM.addEventListener("submit", function (event)
{
    event.preventDefault();

    let assetGroup = REMOVE_INPUT.value;
    let assets = REMOVE_ASSET_INPUT.value

    removeForm(assetGroup, assets);
});

RENAME_FORM.addEventListener("submit", function (event)
{
    event.preventDefault();

    let assetGroup = RENAME_INPUT.value;

    renameForm(assetGroup);
});

DELETE_FORM.addEventListener("submit", function (event)
{
    event.preventDefault();

    let assetGroup = DELETE_INPUT.value

    deleteForm(assetGroup);
});

async function getForm()
{
    if (!sessionStorage.getItem("token"))
    {
        throw new Error("Authentication token: The authentication token is empty");
    }

    try
    {
        console.log("Getting asset groups");
        const ASSET_GROUPS = await getAssetGroups(sessionStorage.getItem("token"));
        console.log(JSON.stringify(ASSET_GROUPS, null, 4));
    }
    catch (error)
    {
        throw new Error(error);
    }
}

async function createForm()
{
    if (!sessionStorage.getItem("token"))
    {
        throw new Error("Authentication token: The authentication token is empty");
    }

    try
    {
        console.log("Creating asset groups");
        const ASSET_GROUPS = await createAssetGroup(sessionStorage.getItem("token"));
        console.log(JSON.stringify(ASSET_GROUPS, null, 4));
    }
    catch (error)
    {
        throw new Error(error);
    }
}

async function addForm(ASSET_GROUP, ASSETS)
{
    if (!sessionStorage.getItem("token"))
    {
        throw new Error("Authentication token: The authentication token is empty");
    }

    try
    {
        console.log("Adding assets to asset group");
        const ASSET_GROUPS = await addAssetToAssetGroup(sessionStorage.getItem("token"), ASSET_GROUP, ASSETS.split(","));
        console.log(JSON.stringify(ASSET_GROUPS, null, 4));
    }
    catch (error)
    {
        throw new Error(error);
    }
}

async function removeForm(ASSET_GROUP, ASSETS)
{
    if (!sessionStorage.getItem("token"))
    {
        throw new Error("Authentication token: The authentication token is empty");
    }

    try
    {
        console.log("Removeing asset groups");
        const ASSET_GROUPS = await removeAssetFromAssetGroup(sessionStorage.getItem("token"), ASSET_GROUP, ASSETS.split(","));
        console.log(JSON.stringify(ASSET_GROUPS, null, 4));
    }
    catch (error)
    {
        throw new Error(error);
    }
}

async function renameForm(ASSET_GROUP)
{
    if (!sessionStorage.getItem("token"))
    {
        throw new Error("Authentication token: The authentication token is empty");
    }

    try
    {
        console.log("Renaming asset groups");
        const ASSET_GROUPS = await renameAssetGroup(sessionStorage.getItem("token"), ASSET_GROUP);
        console.log(JSON.stringify(ASSET_GROUPS, null, 4));
    }
    catch (error)
    {
        throw new Error(error);
    }
}

async function deleteForm(ASSET_GROUP)
{
    if (!sessionStorage.getItem("token"))
    {
        throw new Error("Authentication token: The authentication token is empty");
    }

    try
    {
        console.log("Deleting asset groups");
        const ASSET_GROUPS = await deleteAssetGroup(sessionStorage.getItem("token"), ASSET_GROUP);
        console.log(JSON.stringify(ASSET_GROUPS, null, 4));
    }
    catch (error)
    {
        throw new Error(error);
    }
}

File: nomad-samples/python/asset_groups/asset_groups.py
from asset_group.add_asset_to_asset_group import *
from asset_group.create_asset_group import *
from asset_group.delete_asset_group import *
from asset_group.get_asset_groups import *
from asset_group.remove_asset_from_asset_group import *
from asset_group.rename_asset_group import *

def get_asset_groups_main(AUTH_TOKEN):
    try:
        print("Getting asset groups")
        print(json.dumps(get_asset_groups(AUTH_TOKEN), indent=4))
    except:
        raise Exception()

def create_asset_groups_main(AUTH_TOKEN):
    try:
        ASSET_GROUP_ID = input("Input an asset group id (press enter to skip): ")
        ASSET_GROUP_NAME = input("Input an asset group name (press enter to skip): ")
        ASSETS = input("Enter asset ids you wish to add to an asset group (separate by comma) (press enter to skip): ")

        BODY = {}

        if ASSET_GROUP_ID != "":
            BODY["id"] = ASSET_GROUP_ID
        if ASSET_GROUP_NAME != "":
            BODY["name"] = ASSET_GROUP_NAME
        if ASSETS != "":
            BODY["assets"] = ASSETS.split(",") 

        print("Creating a new asset group") 
        INFO = create_asset_group(AUTH_TOKEN, BODY)
        print(json.dumps(INFO, indent=4))
    except:
        raise Exception()

def add_asset_to_asset_group_main(AUTH_TOKEN):
    try:
        ASSET_GROUP_ID = input("Input an asset group id: ")
        ASSETS = input("Enter asset ids you wish to add to an asset group (separate by comma): ")

        print("Adding asset to asset group")
        INFO = add_asset_to_asset_group(AUTH_TOKEN, ASSET_GROUP_ID, ASSETS.split(","))
        print(json.dumps(INFO, indent=4))
    except:
        raise Exception() 

def remove_asset_from_asset_group_main(AUTH_TOKEN):
    try:
        ASSET_GROUP_ID = input("Input an asset group id: ")
        ASSETS = input("Enter asset ids you wish to remove from an asset group (separate by comma): ")

        print("Removing asset from asset group")
        INFO = remove_asset_to_asset_group(AUTH_TOKEN, ASSET_GROUP_ID, ASSETS.split(","))
        print(json.dumps(INFO, indent=4))
    except:
        raise Exception()

def rename_asset_group_main(AUTH_TOKEN):
    try: 
        ASSET_GROUP_ID = input("Input an asset group id: ")
        ASSET_GROUP_NEW_NAME = input("Input a new name for an asset group: ")

        print("Renaming asset group")
        INFO = rename_asset_group(AUTH_TOKEN, ASSET_GROUP_ID, ASSET_GROUP_NEW_NAME)
        print(json.dumps(INFO, indent=4))
    except:
        raise Exception()

def delete_asset_group_main(AUTH_TOKEN):
    try:
        ASSET_GROUP_ID = input("Input an asset group id: ")

        print("Deleting asset group")
        INFO = delete_asset_group(AUTH_TOKEN, ASSET_GROUP_ID)
        print(json.dumps(INFO, indent=4))
    except:
        raise Exception()
    


if __name__ == "__main__":
    AUTH_TOKEN = input("Enter your authentication token: ")
    
    while True:
        print("Do you want to get your asset groups, create an asset group, add an asset to an "\
              "asset group, remove an asset from an asset group, rename an asset, delete an asset, "\
              "or exit")
        USER_INPUT = input("Enter get, create, add, remove, rename, delete, or exit for each option above respectively: ")

        if USER_INPUT == "get":
            get_asset_groups_main(AUTH_TOKEN)
        elif USER_INPUT == "create":
            create_asset_groups_main(AUTH_TOKEN)
        elif USER_INPUT == "add":
            add_asset_to_asset_group_main(AUTH_TOKEN)
        elif USER_INPUT == "remove":
            remove_asset_from_asset_group_main(AUTH_TOKEN)
        elif USER_INPUT == "rename":
            rename_asset_group_main(AUTH_TOKEN)
        elif USER_INPUT == "delete":
            delete_asset_group_main(AUTH_TOKEN)
        elif USER_INPUT == "exit":
            break
        else:
            print("Input invalid")
        
